### PR TITLE
Bump mavenPublish to 0.15.1

### PR DESCRIPTION
**Pull request by Goooler**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -53,7 +53,7 @@ jobs:
           java-version: 14
 
       - name: Upload Artifacts
-        run: ./gradlew uploadArchives
+        run: ./gradlew publish --no-daemon --no-parallel
```
Comment (JakeWharton) [inline .github/workflows/build.yml:56]: The daemon is already disabled via `GRADLE_OPTS` and we do not enable parallel in `gradle.properties` so neither should have an effect.

**Review by JakeWharton — APPROVED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (Goooler) [reply .github/workflows/build.yml]: Modified

**Review by Egorand — APPROVED** [review]
Perfect, thank you!

**Comment by Egorand** [comment]
@Goooler the build failed: https://github.com/square/kotlinpoet/runs/2830574581?check_suite_focus=true

Should

```
ORG_GRADLE_PROJECT_mavenCentralRepositoryUsername: ${{ secrets.SONATYPE_NEXUS_USERNAME }}          ORG_GRADLE_PROJECT_mavenCentralRepositoryPassword: ${{ secrets.SONATYPE_NEXUS_PASSWORD }}
```

in `build.yml` be changed to

```
ORG_GRADLE_PROJECT_ mavenCentralUsername: ${{ secrets.SONATYPE_NEXUS_USERNAME }}          ORG_GRADLE_PROJECT_mavenCentralPassword: ${{ secrets.SONATYPE_NEXUS_PASSWORD }}
```

?

**Comment by Goooler** [comment]
> @Goooler the build failed: https://github.com/square/kotlinpoet/runs/2830574581?check_suite_focus=true
> 
> Should
> 
> ```
> ORG_GRADLE_PROJECT_mavenCentralRepositoryUsername: ${{ secrets.SONATYPE_NEXUS_USERNAME }}
> ORG_GRADLE_PROJECT_mavenCentralRepositoryPassword: ${{ secrets.SONATYPE_NEXUS_PASSWORD }}
> ```
> 
> in `build.yml` be changed to
> 
> ```
> ORG_GRADLE_PROJECT_ mavenCentralUsername: ${{ secrets.SONATYPE_NEXUS_USERNAME }}
> ORG_GRADLE_PROJECT_mavenCentralPassword: ${{ secrets.SONATYPE_NEXUS_PASSWORD }}
> ```
> 
> ?

Yeah, sorry I forget to modify that

**Pull request merged**